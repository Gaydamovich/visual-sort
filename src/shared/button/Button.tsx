import React, { FC } from "react";
import cn from "classnames";

import styles from "./Button.module.css";

interface ButtonProps {
  text: string;
  className?: string;
  onClick: () => void;
}

export const Button: FC<ButtonProps> = ({ text, className, onClick }) => (
  <button className={cn(styles.btn, className)} onClick={onClick}>
    {text}
  </button>
);
