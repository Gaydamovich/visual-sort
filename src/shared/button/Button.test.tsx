import { render } from "@testing-library/react";

import { Button } from "./Button";

describe("Testing Button component", () => {
  const ButtonComponent = <Button text="Кнопка" onClick={() => {}} />;

  it("Get snapshot Button component", () => {
    const { asFragment } = render(ButtonComponent);
    expect(asFragment()).toMatchSnapshot();
  });
});
