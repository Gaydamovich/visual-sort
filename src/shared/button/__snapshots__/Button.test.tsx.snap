// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Get snapshot Button component 1`] = `
<DocumentFragment>
  <button
    class="btn"
  >
    Кнопка
  </button>
</DocumentFragment>
`;
